src/main/java/io/openshift/booster/catalog/BoosterCatalogService.java

```diff
@@ -344,27 +357,27 @@ public Path copy(Booster booster, Path projectRoot) throws IOException
    }
 
    @Override
-   public Optional<Booster> getBooster(Mission mission, Runtime runtime)
+   public Optional<Booster> getBooster(Mission mission, Runtime runtime, String ...labels)
    {
-      return getBooster(mission, runtime, null);
+      return getBooster(mission, runtime, null, labels);
    }
 
    @Override
-   public Optional<Booster> getBooster(Mission mission, Runtime runtime, Version version)
+   public Optional<Booster> getBooster(Mission mission, Runtime runtime, Version version, String ...labels)
    {
       Objects.requireNonNull(mission, "Mission should not be null");
       Objects.requireNonNull(runtime, "Runtime should not be null");
-      return boosters.stream()
+      return filtered(boosters.stream(), labels)
                .filter(b -> mission.equals(b.getMission()))
                .filter(b -> runtime.equals(b.getRuntime()))
                .filter(b -> version == null || version.equals(b.getVersion()))
                .findFirst();
    }
 
    @Override
-   public Collection<Booster> getBoosters()
+   public Collection<Booster> getBoosters(String ...labels)
    {
-      return Collections.unmodifiableCollection(boosters);
+      return filtered(boosters.stream(), labels).collect(Collectors.toCollection(HashSet::new));
```

Comment:
Even though it's a copy, the returned collection is still modifiable

Reply:
Right.  But the rest of the api methods return modifiable collections don't they?  Seems like this change make the API more consistent.

Reply:
You're right. They should all return immutable collections. We need to fix that